gst/tensor_merge/gsttensormerge.c

```diff
@@ -711,6 +748,11 @@ gst_tensor_merge_collected (GstCollectPads * pads,
     tensor_merge->negotiated = TRUE;
   }
 
+  if (tensor_merge->need_buffer) {
```

Comment:
How about moving these lines to line 729 (before caps-negotiated, 'need_buffer' would be set true before getting all pads info.)

Reply:
How about changing code to unref tensors_buf
gst_buffer_unref (tensors_buf); in this code block, not in gst_tensor_merge_collect_buffer() line 511

Reply:
Yes. I would be good!. Done!